operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -837,17 +840,29 @@ public void onThrowable(Packet packet, Throwable throwable) {
                   }
                 });
 
-            gate.getExecutor()
-                .schedule(
-                    () -> {
-                      DomainPresenceInfo existing = getExistingDomainPresenceInfo(ns, domainUID);
-                      if (existing != null) {
-                        existing.setPopulated(false);
-                        makeRightDomainPresence(existing, true, isDeleting, false);
-                      }
-                    },
-                    DomainPresence.getDomainPresenceFailureRetrySeconds(),
-                    TimeUnit.SECONDS);
+            DomainPresenceInfo existing = getExistingDomainPresenceInfo(ns, domainUID);
+            if (existing != null) {
+              int failureCount = existing.incrementAndGetFailureCount();
+              LOGGER.fine(
+                  "Failure count for DomainPresenceInfo: " + existing + " is now: " + failureCount);
+              if (failureCount > DomainPresence.getDomainPresenceFailureRetryMaxCount()) {
+                LOGGER.warning(
+                    MessageKeys.CANNOT_START_DOMAIN_AFTER_MAX_RETRIES,
+                    domainUID,
+                    ns,
+                    DomainPresence.getDomainPresenceFailureRetryMaxCount(),
+                    throwable);
+              } else {
+                gate.getExecutor()
+                    .schedule(
+                        () -> {
+                          existing.setPopulated(false);
```

Comment:
This changes the behavior incorrectly...  We need to get the existing domain presence at the time when the scheduled event happens.  You've moved the lookup to when the retry is scheduled.